extensions/standard-processors/processors/ReplaceText.cpp

```diff
@@ -0,0 +1,342 @@
+    ->withDescription("Run the 'Replacement Strategy' against each line separately (Line-by-Line) for All lines in the FlowFile, "
+                      "First Line (Header) only, Last Line (Footer) only, all Except the First Line (Header) or all Except the Last Line (Footer).")
+    ->isRequired(false)
+    ->withDefaultValue<std::string>(toString(LineByLineEvaluationModeType::ALL))
+    ->withAllowableValues(LineByLineEvaluationModeType::values())
+    ->build();
+
+const core::Property ReplaceText::ReplacementStrategy = core::PropertyBuilder::createProperty("Replacement Strategy")
+    ->withDescription("The strategy for how and what to replace within the FlowFile's text content. "
+                      "Substitute Variables replaces ${attribute_name} placeholders with the corresponding attribute's value "
+                      "(if an attribute is not found, the placeholder is kept as it was).")
+    ->isRequired(true)
+    ->withDefaultValue(toString(ReplacementStrategyType::REGEX_REPLACE))
+    ->withAllowableValues(ReplacementStrategyType::values())
+    ->build();
+
+const core::Property ReplaceText::SearchValue = core::PropertyBuilder::createProperty("Search Value")
+    ->withDescription("The Search Value to search for in the FlowFile content. "
+                      "Only used for 'Literal Replace' and 'Regex Replace' matching strategies. "
+                      "Supports expression language except in Regex Replace mode.")
+    ->isRequired(false)
+    ->supportsExpressionLanguage(true)
+    ->build();
+
+const core::Property ReplaceText::ReplacementValue = core::PropertyBuilder::createProperty("Replacement Value")
+    ->withDescription("The value to insert using the 'Replacement Strategy'. "
+                      "Using 'Regex Replace' back-references to Regular Expression capturing groups are supported: "
+                      "$& is the entire matched substring, $1, $2, ... are the matched capturing groups. Use $$1 for a literal $1. "
+                      "Back-references to non-existent capturing groups will be replaced by empty strings. "
+                      "Supports expression language except in Regex Replace mode.")
+    ->isRequired(true)
+    ->supportsExpressionLanguage(true)
+    ->build();
+
+const core::Relationship ReplaceText::Success("success", "FlowFiles that have been successfully processed are routed to this relationship. "
+                                                         "This includes both FlowFiles that had text replaced and those that did not.");
+const core::Relationship ReplaceText::Failure("failure", "FlowFiles that could not be updated are routed to this relationship.");
+
+ReplaceText::ReplaceText(const std::string& name, const utils::Identifier& uuid)
+  : core::Processor(name, uuid),
+    logger_(logging::LoggerFactory<ReplaceText>::getLogger()) {
+}
+
+core::annotation::Input ReplaceText::getInputRequirement() const {
+  return core::annotation::Input::INPUT_REQUIRED;
```

Comment:
all other `getInputRequirement` overrides are implemented in the headers, which I kind of agree with, as its implementation (the return value) is something inherent to the processor class (like the relationships and properties)

Reply:
yup, makes sense

done in https://github.com/apache/nifi-minifi-cpp/pull/1170/commits/fead017bb2845854247d04c64d51882b273aba5a